### PR TITLE
bpo-39802: Only implement set_escdelay and set_tabsize when curses extensions are activated

### PR DESCRIPTION
**Pull request by isidentical**
https://bugs.python.org/issue39802

### CODE REVIEW

File: Modules/_cursesmodule.c [hunk 1]
```diff
@@ -3255,6 +3255,7 @@ _curses_setupterm_impl(PyObject *module, const char *term, int fd)
     Py_RETURN_NONE;
 }
 
+#if defined(NCURSES_EXT_FUNCS) && NCURSES_EXT_FUNCS >= 20081102
```
Comment (pablogsal) [inline Modules/_cursesmodule.c:3258]: Can you add a short comment explaining where `20081102` comes from?

**Review by pablogsal — APPROVED** [review]
LGTM modulo the nit about the comment

**Comment by pablogsal** [comment]
Will do a buildbot check and merge if everything goes good :)

**Comment by pablogsal** [comment]
@isidentical  Seems that the refleaks buildbots do not like this PR

**Comment by pablogsal** [comment]
> @isidentical Seems that the refleaks buildbots do not like this PR :(

That was unrelated stuff that was just fixed in https://bugs.python.org/issue39831

**Review by pablogsal — APPROVED** [review]

**Comment by pablogsal** [comment]
@isidentical Could you work on the backports?

**Comment by isidentical** [comment]
IIRC these functions added in 3.9, so no need to do a cherrypick

**Pull request merged**